compiler/rustc_trait_selection/src/solve/eval_ctxt/canonical.rs

```diff
@@ -287,4 +303,29 @@ impl<'tcx> EvalCtxt<'_, 'tcx> {
             let _ = member_constraint;
         }
     }
+
+    fn register_opaque_types(
+        &mut self,
+        param_env: ty::ParamEnv<'tcx>,
+        opaque_types: &[(ty::OpaqueTypeKey<'tcx>, Ty<'tcx>)],
+    ) -> Result<Vec<Goal<'tcx, ty::Predicate<'tcx>>>, NoSolution> {
+        let mut nested_goals = vec![];
+        for &(a, b) in opaque_types {
+            nested_goals.extend(
+                self.infcx
+                    .handle_opaque_type(
+                        self.tcx().mk_opaque(a.def_id.to_def_id(), a.substs),
+                        b,
+                        true,
+                        &ObligationCause::dummy(),
+                        param_env,
+                    )?
+                    .into_obligations()
```

Comment:
i think we should throw away obligation here, we already try to prove them inside of the query we're just returned from.

pls add a comment

Reply:
I am convinced -- I'm just gonna throw away these obligations.